server/controllers/appointments/editAppointmentStatus/index.js

```diff
@@ -1,38 +1,44 @@
-const { boomify, appointmentIdValidation } = require('../../../utils');
+const { boomify, appointmentStatusValidation } = require('../../../utils');
 
-const {
-  updateAppointmentStatusQuery,
-  getAppointmentsStatusByIdQuery,
-} = require('../../../database/queries');
+const { updateAppointmentStatusQuery } = require('../../../database/queries');
 
 const editAppointmentStatus = async (req, res, next) => {
   try {
-    const { appointmentId } = await appointmentIdValidation.validate(
-      req.params,
-    );
-
     const {
-      rows: appointmentById,
-      rowCount: isAppointmentExist,
-    } = await getAppointmentsStatusByIdQuery({
       appointmentId,
-    });
+      isDone,
+    } = await appointmentStatusValidation.validate(
+      {
+        ...req.params,
+        ...req.body,
+      },
+      { abortEarly: false },
+    );
 
-    if (!isAppointmentExist) {
+    if (isDone === true)
       return next(
-        boomify(400, 'Invalid Appointment id', 'This appointment is not exist'),
+        boomify(400, 'Closed Appointment', 'This appointment is completed'),
       );
-    }
-    const { is_done: isAppointmentDone } = appointmentById[0];
-
-    if (isAppointmentDone) {
+    if (isDone !== false)
       return next(
-        boomify(400, 'Closed Appointment', 'This appointment is completed'),
+        boomify(
+          409,
+          'Unavailable Time',
+          'please choose another appointment time',
+        ),
       );
```

Comment:
Since you are already checking for `isDone` value to be `true/false` value only any you checked if `isDone=true` so there's only one possible value which is `false` so this check is unnecessary

Reply:
ok